src/index.js

```diff
@@ -1,27 +1,23 @@
 export default function createExtraScopePlugin(...extra) {
-  const scopes = extra.map(scope => scope.trim())
+  const scopes = extra.map((scope) => `${scope.trim()} `)
 
-  const extraScopePlugin = (
-    context,
-    content,
-    selectors,
-    parents,
-    line,
-    column,
-    length,
-    type,
-  ) => {
-    if (context !== -1) {
+  return (element) => {
+    if (element.type !== 'rule') {
       return
     }
 
-    const selector = selectors[0]
-    selectors[0] = `${scopes[0]} ${selector}`
+    if (element.root?.type === '@keyframes') {
+      return
+    }
 
-    for (let i = 1; i < scopes.length; i++) {
-      selectors.push(`${scopes[i]} ${selector}`)
+    if (
+      !element.parent ||
+      (element.props.length === 1 && element.value.charCodeAt(0) !== 58) ||
+      !element.length
+    ) {
+      element.props = element.props
+        .map((prop) => scopes.map((scope) => scope + prop))
+        .reduce((scopesArray, scope) => scopesArray.concat(scope), [])
     }
```

Comment:
I think that this if statement here doesn't make sense, at least for the most part. Especially parts focusing on anything else than `element.parent` seems dubious - they are trying to battle the logic from the compat plugin, right?

If I just apply this change:
```suggestion
    element.props = element.props
      .map((prop) => scopes.map((scope) => scope + prop))
      .reduce((scopesArray, scope) => scopesArray.concat(scope), [])
```

then "vanilla" tests still pass and that is good. We, of course, still need to figure out how to fix the compatibility with compat plugin though

Reply:
q: what if we'd serialize the current rule node and set its `.return` property with the scope prefixed to the output? maybe we don't have to manipulate `element.props` at all?

Reply:
Yeah, this change makes sense to me. It matches up with @tpict's [original implementation](https://github.com/Andarist/stylis-plugin-extra-scope/pull/11/commits/6c267e09da40740efeb93e274c491716f344bb6a), before the changes to account for `compat`.

Reply:
hmm, the serializing advice could work out if the received `callback` would be pure but in the case of Emotion it contains the `rulesheet` plugin so we can't just call into it freely

Maybe worth opening a discussion in the Stylis problem about this? I'm trying to think through how those 2 plugins can coexist but so far I didn't get an epiphany with the solution 😬

Reply:
Sorry, commented before seeing that second message. Can you explain more how the `.return` prop is used? Are any good docs for writing plugins outside of the [examples in Middleware.js](https://github.com/thysultan/stylis.js/blob/master/src/Middleware.js)?

Reply:
Well, this is a little bit fuzzy to me but IIRC `.return` holds the "final" stringified value of a given node. I think though that this is a little bit over my head - it would be best to create a discussion in the Stylis repo so maybe we could get some guidance from Sultan for this.

Reply:
Alright, just started a discussion about it in Stylis: https://github.com/thysultan/stylis.js/discussions/281